### ISSUE TITLE
Projectiles sometimes follow weird paths after collisions

### ISSUE THREAD

**Issue by jaloisi834** [jaloisi834/operation-space-cow#13]
I noticed that after a projectile hits an object in certain spots (seems to be near corners) it will change path slightly. For example, (assume 0 degrees is to the right and 90 degrees is up) if the projectile is going at about a 45 degree angle and it hits the corner of an object, it can change to something around a 60 degree angle.

**Comment by jaloisi834** [comment]
After thinking about this more, it seems that this behavior alone is unavoidable since it is technically intended. When an actor hits a wall at an angle, it can't move in the direction of the wall anymore, but will continue moving on the other axis until it hits something else. In the case of a projectile, its activeDirection will keep it moving towards its goal when it hits a wall. The acceleration is responsible for the bending effect as it comes off of the corner because the projectile is accelerating from a stop on that axis.

This all being said, obviously it still looks bad to have the projectile curving all weird. This didn't seem to be happening when projectiles were able to delete themselves right away, so I think that change may have something to do with this.

**Comment by jaloisi834** [comment]
This must be tied to issue #1 regarding actors bouncing off of corners. There is clearly an issue with the collision detection under these conditions. I tested this by adding a print statement in Projectile's reactToCollision() function. Normally, when a projectile hits something, its reactToCollision() function would be called and it would be deleted. Instead, I found that when shooting at an object's corner there were no statements printed, which means the projectile was being stopped at it's maximum range instead.
